erpnext/hr/doctype/job_offer/job_offer.py

```diff
@@ -24,8 +24,13 @@ def validate_vacancies(self):
 		check_vacancies = frappe.get_single("HR Settings").check_vacancies
 		if staffing_plan and check_vacancies:
 			job_offers = self.get_job_offer(staffing_plan.from_date, staffing_plan.to_date)
-			if staffing_plan.vacancies - len(job_offers) <= 0:
-				frappe.throw(_("There are no vacancies under staffing plan {0}").format(frappe.bold(get_link_to_form("Staffing Plan", staffing_plan.parent))))
+
+			if not staffing_plan.get("vacancies") or staffing_plan.vacancies - len(job_offers) <= 0:
+				error_variable = 'for ' + frappe.bold(self.designation)
+				if staffing_plan.parent:
```

Comment:
```suggestion
				if staffing_plan.get("parent"):
```
If no vacancies high chances are that parent is None too. Try to use get in most places where it is uncertain if the attribute exists